src/routes/routerConfig.tsx

```diff
@@ -26,60 +15,60 @@ import Welcome from '../pages/welcome/Welcome';
 
 // Conditionally allow access to route via `loader` instead of conditionally defining these routes in react-router v6.4.
 // See https://github.com/remix-run/react-router/discussions/10223#discussioncomment-5909050
-const conditionalLoader = (condition: boolean, redirectTo: string) => () => {
+const conditionalLoader = (condition: boolean, redirectTo: string, returnValue?: any) => () => {
   if (condition) {
     return redirect(redirectTo);
   }
-  return null;
+  return returnValue ?? null;
 };
 
 export const getDisabledRouterConfig: (reason: string | boolean) => RouteObject[] = reason => {
-  const disabledElement = <Disabled reason={typeof reason === 'string' ? reason : undefined} />;
-
+  const disabledReason = typeof reason === 'string' ? reason : undefined;
   return [
     {
       path: '/*',
       element: <Application />,
       children: [
         {
           path: 'login',
-          element: <Login />,
+          lazy: () => import('../pages/login/Login'),
           loader: conditionalLoader(Constants.playgroundOnly, '/')
         },
         {
           path: '',
-          element: disabledElement,
-          loader: conditionalLoader(!Constants.playgroundOnly, 'login')
+          lazy: () => import('../pages/disabled/Disabled'),
+          loader: conditionalLoader(!Constants.playgroundOnly, 'login', disabledReason)
         },
         {
           path: 'courses/*',
           element: <Navigate to="/login" />
         },
         {
           path: '*',
-          element: disabledElement
+          lazy: () => import('../pages/disabled/Disabled'),
+          loader: () => disabledReason
         }
       ]
     }
   ];
 };
 
-const commonChildrenRoutes = [
+const commonChildrenRoutes: RouteObject[] = [
   {
     path: 'contributors',
-    element: <Contributors />
+    lazy: () => import('../pages/contributors/Contributors')
   },
   {
     path: 'callback/github',
-    element: <GitHubCallback />
+    lazy: () => import('../pages/githubCallback/GitHubCallback')
   },
   {
     path: 'sicpjs/:section?',
-    element: <Sicp />
+    lazy: () => import('../pages/sicp/Sicp')
   },
   {
     path: 'githubassessments/*',
-    element: <GitHubClassroom />,
+    lazy: () => import('../pages/githubAssessments/GitHubClassroom'),
```

Comment:
Would it be better if we define all of the imports at the top-level, for better readibility? E.g.

```tsx
const Contributors = () => import('../pages/contributors/Contributors');
const GitHubCallback = () => import('../pages/githubCallback/GitHubCallback');
const Sicp = () => import('../pages/sicp/Sicp');
const GitHubClassroom = () => import('../pages/githubAssessments/GitHubClassroom');
// and so on...
```

and then, later on:

```tsx
{
  // ...
  lazy: Contributors // or GitHubCallback or Sicp or...
}
```

Reply:
yup good idea, was deciding whether to do this haha, but did not since these imports are mostly used once

will update this tmr!